docs/rfcs/rfc-034-auth.md

```diff
@@ -0,0 +1,859 @@
+# Authentication and Authorization
+
+## API
+
+### Instantiation
+
+Authentication and authorization features are currently exposed by a browser-incompatible plugin.
+
+As part of our efforts to simplify interaction with the library and allow users to experiment with features in the GraphQL Toolbox, we should strive to once again build these features directly into the library.
+
+This will be achievable by substituting our use of [jsonwebtoken](https://www.npmjs.com/package/jsonwebtoken) for [jose](https://www.npmjs.com/package/jose). It has been validated that `jose` will be able to decode and verify tokens as we currently do, and it is browser compatible.
+
+#### Proposal
+
+The following is a type proposal for auth configuration:
+
+```ts
+type JWKS = {
+  uri: string;
+}
+
+type AuthConfig = {
+  secret: string | JWKS
+  verify: boolean;
+  jwtPayload: Schema;
+}
+```
+
+`AuthConfig.secret` will assume symmetric type if passed a string, or other types by passing in different objects. This will allow us to use the same configuration for a variety of secret types. `jose` also supports SPKI encoded RSA keys.
+
+`AuthConfig.verify` will be `true` by default, but can be set to `false` if the desired behaviour is to decode only.
```

Comment:
Can we consider defining `verify`/`jwtPayload` on a schema directive so that this information is defined in the same place it is relevant?

Reply:
I'm not sure about this... `jwtPayload` has some nice TypeScript types which we could leverage - I think this is just too complex to put into type definitions.